ApiDoctor.Publishing/CSDL/csdlwriter.cs

```diff
@@ -1851,8 +1844,164 @@ private static void AddReadByKeyRestriction(IODataAnnotatable annotatable, Metho
                 }
             }
         }
-    }
 
+        private static Dictionary<string, string> httpMethodToRestrictionTermMappings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+           { "GET", Term.ReadRestrictionTerm },
+           { "POST", Term.InsertRestrictionsTerm },
+           { "PATCH", Term.UpdateRestrictionsTerm },
+           { "PUT", Term.UpdateRestrictionsTerm },
+           { "DELETE", Term.DeleteRestrictionsTerm }
+        };
+
+        private static string GetRestrictionTermForMethod(string httpMethod)
+        {
+            httpMethodToRestrictionTermMappings.TryGetValue(httpMethod, out string restrictionTerm);
+            return restrictionTerm;
+        }
+
+
+#region Links Annotations
+        private static string CreateHrefValue(string sourceFilePath)
+        {
+
+            var uriBuilder = new UriBuilder()
+            {
+                Scheme = "https",
+                Host = "docs.microsoft.com",
+                Path = $@"graph{sourceFilePath}",
+                Query = "?view=graph-rest-1.0"
```

Comment:
I believe its possible to have the docs point to beta. If the API exists in beta not v1.0 that may pose a problem.
Maybe we can infer this from the location of the source method doc file?